staticd/static_nb_config.c

```diff
@@ -152,6 +152,14 @@ static bool static_nexthop_create(struct nb_cb_create_args *args,
 				return NB_ERR_VALIDATION;
 			}
 		}
+		nh_type = yang_dnode_get_enum(args->dnode, "./nh-type");
+		nh_vrf = yang_dnode_get_string(args->dnode, "./vrf");
+		if (nh_type == STATIC_BLACKHOLE && nh_vrf) {
```

Comment:
vrf is part of the nexthop key, it is always set. this check is always `true` for blackhole routes.

Reply:
you need to compare nexthop's vrf with the route's vrf to achieve what you want.